sycamore/transforms/embed.py

```diff
@@ -90,7 +96,7 @@ def generate_embeddings(self, doc_batch: list[Document]) -> list[Document]:
 
         assert self._transformer is not None
 
-        text_batch = ["" if doc.text_representation is None else doc.text_representation for doc in doc_batch]
+        text_batch = [self.pre_process_document(doc) for doc in doc_batch if doc.text_representation is not None]
         embeddings = self._transformer.encode(text_batch, batch_size=self.model_batch_size, device=self.device)
         for doc, embedding in zip(doc_batch, embeddings):
```

Comment:
well now doc_batch and embeddings could be different lengths bc maybe a doc got take out of the middle, so won't that cause potential errors?

Reply:
good catch! Updated